core/src/main/java/org/sonarsource/sonarlint/core/http/MutableTrustManager.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.http;
+
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.CheckForNull;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import nl.altindag.ssl.util.CertificateUtils;
+import nl.altindag.ssl.util.KeyStoreUtils;
+import org.apache.commons.io.FileUtils;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+/**
+ * Inspired by
+ * https://github.com/JetBrains/intellij-community/blob/f4fc17b0c44d38d65fb7ad47b968bed55c889609/platform/platform-api/src/com/intellij/util/net/ssl/ConfirmingTrustManager.java#L313
+ * 
+ */
+public final class MutableTrustManager implements X509TrustManager {
+
+  private static final X509Certificate[] NO_CERTIFICATES = new X509Certificate[0];
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+  private final Path myPath;
+  private final String myPassword;
```

Comment:
I think it's not necessary to store the password as an attribute, can be passed in the constructor once to create the key store, and then we can pass the password when adding a certificate. It would avoid having it in the memory, and it shouldn't be a big performance issue to read the system property only when adding a certificate. Also, the password should be stored as char[] for security purposes IMO

WDYT?

Reply:
I don't see how we can avoid having the password in memory at some point.
I am not aware that `char[]` is more secure than a String. Do you have any pointers to that?

Reply:
Also I think the goal is to not create an empty keystore as long as a custom certificate has not been approved

Reply:
OK, interesting to know, but I don't think this is a concern for us, as the password is a constant anyway.